src/components/tl/ucp/allreduce/allreduce.h

```diff
@@ -77,6 +78,16 @@ ucc_status_t ucc_tl_ucp_allreduce_dbt_start(ucc_coll_task_t *task);
 
 ucc_status_t ucc_tl_ucp_allreduce_dbt_progress(ucc_coll_task_t *task);
 
+void ucc_tl_ucp_allreduce_ring_progress(ucc_coll_task_t *coll_task);
+
+ucc_status_t ucc_tl_ucp_allreduce_ring_start(ucc_coll_task_t *coll_task);
```

Comment:
Advice: Consider not exposing functions in the public API if they are not intended for use outside the algorithm's internal logic. Keeping the API surface minimal helps with maintainability and reduces potential misuse.